cognite/client/utils/_token_generator.py

```diff
@@ -29,15 +32,16 @@ def __init__(
             self._access_token_expires_at = None
 
     def return_access_token(self) -> str:
-        if not self.token_params_set():
-            raise CogniteAPIKeyError("Could not generate access token - missing token generation arguments")
-        elif self._access_token is None:
-            raise CogniteAPIKeyError("Could not generate access token from provided token generation arguments")
+        with LOCK:
```

Comment:
Great catch! But we instantiate one TokenGenerator per CogniteClient, so I believe this lock should be scoped to the instance.